WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m

```diff
@@ -41,6 +62,52 @@ - (BOOL)fb_goToHomescreenWithError:(NSError **)error
   return YES;
 }
 
+- (BOOL)fb_lockScreen:(NSError **)error
+{
+  if (fb_isLocked) {
+    return YES;
+  }
+  SEL mSelector = NSSelectorFromString(@"pressLockButton");
+  [self methodSignatureForSelector:mSelector];
+  NSMethodSignature *mSignature = [self methodSignatureForSelector:mSelector];
+  NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:mSignature];
+  [invocation setTarget:self];
+  [invocation setSelector:mSelector];
+  [invocation invoke];
+  return [[[[FBRunLoopSpinner new]
+            timeout:FBScreenLockTimeout]
+           timeoutErrorMessage:@"Timed out while waiting until the screen gets locked"]
+          spinUntilTrue:^BOOL{
+            return fb_isLocked;
+          } error:error];
+}
+
+- (BOOL)fb_isScreenLocked
+{
+  return fb_isLocked;
+}
+
+- (BOOL)fb_unlockScreen:(NSError **)error
+{
+  if (!fb_isLocked) {
+    return YES;
+  }
+  [self pressButton:XCUIDeviceButtonHome];
+  [[NSRunLoop currentRunLoop] runUntilDate:[NSDate dateWithTimeIntervalSinceNow:FBHomeButtonCoolOffTime]];
```

Comment:
I don't think you need this. Usually higher XCUITest apis already have this cool off implemented or have you seen a case that this was needed?

Reply:
there is a comment in the code that it was the case in 9.3. I'd rather keep the delay, since there is anyway an animation delay there and XCTest does not really like animations.